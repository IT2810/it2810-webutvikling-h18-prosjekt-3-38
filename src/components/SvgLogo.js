import React from 'react'
import { View, Animated } from 'react-native'

// Inspiration from https://www.youtube.com/watch?v=vzPmI0GCDPM
class ImageLoader extends React.Component {
  state = {
    opacity: new Animated.Value(0)
  }

  // The useNativeDriver means that the animation happens off the JS thread
  onLoad = () => {
    Animated.timing(this.state.opacity, {
      toValue: 1,
      duration: 500,
      useNativeDriver: true
    }).start()
  }

  // Returns the animated image. ... is the spread operator, which in turn "sends" the props from the
  // SvgLogo to ImageLoader.
  render () {
    return (
      <Animated.Image
        onLoad={this.onLoad}
        {...this.props}
        style={[
          {
            opacity: this.state.opacity,
            transform: [
              {
                scale: this.state.opacity.interpolate({
                  inputRange: [0, 1],
                  outputRange: [0.85, 1]
                })
              }
            ]
          },
          this.props.style
        ]}
      />
    )
  }
}
export default class SvgLogo extends React.Component {
  render () {
    return (
      <View resizeMode='cover' >
        <ImageLoader fadeDuration={0} style={{
          alignSelf: 'center', transform: [{ scale: 0.07 }]
        }} source={require('./img/LogoMoodWalker.jpg')} resizeMode='cover' />
      </View>
    )
  }
}
