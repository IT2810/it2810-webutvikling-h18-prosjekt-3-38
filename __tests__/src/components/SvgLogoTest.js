import React from 'react'
import renderer from 'react-test-renderer'
import SvgLogo from '../../../src/components/SvgLogo.js'

test('renders correctly', () => {
  const tree = renderer.create(<SvgLogo />).toJSON()
  expect(tree).toMatchSnapshot()
})
