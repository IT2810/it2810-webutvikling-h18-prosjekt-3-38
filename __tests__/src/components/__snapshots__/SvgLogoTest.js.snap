// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  resizeMode="cover"
>
  <Image
    fadeDuration={0}
    onLoad={[Function]}
    resizeMode="cover"
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "opacity": 0,
        "transform": Array [
          Object {
            "scale": 0.07,
          },
        ],
      }
    }
  />
</View>
`;
